### PR TITLE
Create target model docs

### PR DESCRIPTION
**Pull request by alexanderahn**
---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1207642546490606

### CODE REVIEW

File: docs/SQL/development/target-models/sql-query.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+This is useful if you import your own codebase and your final query has additional conditions. Your query is saved here for you to view and edit. For low code users you can use a Filter Gem to achieve the same results.
+
+## Enable custom SQL query
+
+You can add a custom SQL Query if you're doing data processing with advance Jinja or dbt templating.
+This gives you flexibility when doing last mile operations for your SQL models. To that end, we support the use of declared variables directly in the SQL queries for those last mile operations on each model write.
+
+### Ports
+
+Ports represent Gem or generated code functions inputs and outputs. The port name is a variable name in Python and Scala code, and table alias in the SQL code. The visual order of the ports defines the order of the variables.
```
Comment (bandikishores) [inline docs/SQL/development/target-models/sql-query.md:24]: I think this is not applicable in SQL.
`Ports represent Gem or generated code functions inputs and outputs. The port name is a variable name in Python and Scala code, and table alias in the SQL code.`

This part in it is still applicable, `table alias in the SQL code`. We can say Port represents the tables or gems that user wants to use in the SQL Query. They can access them using the aliases (port names).

File: docs/SQL/development/target-models/type-and-format.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+- **Table**: Table models are rebuilt as a table on each run. Tables are fast to query, but they can take a long time to rebuild. Use the table materialization type for any models that are queried by BI tools. To apply a write mode to your Target Model, you must use the Table type.
+- **Ephemeral**: Ephemeral models aren't built into the database. Instead, the model code is taken from an ephemeral model and inserted into its dependent models using a common table expression (CTE). Use the ephemeral materialization type for undemanding transformations that appear at the start of your DAG.
+- **Materialized View**: Materialized View models are a combination of a view and a table. They are similar to incremental models in that they serve similar use cases. A materialized view model allows you to create and maintain a materialized view in the target database Warehouse type.
```
Comment (bandikishores) [inline docs/SQL/development/target-models/type-and-format.md:25]: Can we say this is not supported in Snowflake

File: docs/SQL/development/target-models/write-options.md [hunk 3]
```diff
@@ -0,0 +1,306 @@
...
+
+Target Models are considered incremental models, which use a materialization strategy to update your data warehouse tables by loading and transforming just the changed data since the last run. Incremental models avoid processing your entire Dataset, which allows them to update only the new rows. This greatly reduces the time and resources you need to dedicate for your data transformations.
+
+## Write Modes
+
+You can overwrite the target data entirely, append to existing target data, or merge to address slowly changing dimension scenarios.
+
+- To select write modes, you must set the Target Model [Type & Format](type-and-format.md) to **Table**.
+
+Once you've selected your write mode, you'll see in the Code view that the table is stored as a `"materialized": "incremental",` table, with `"incremental_strategy:` set to whichever write mode and merge approach you choose.
```
Comment (bandikishores) [inline docs/SQL/development/target-models/write-options.md:24]: nit - can we remove the comma in the end of code section for `"materialized": "incremental",`

**Review by bandikishores — APPROVED** [review]

File: docs/SQL/development/target-models/sql-query.md (hunk 1 above)
Comment (alexanderahn) [reply docs/SQL/development/target-models/sql-query.md:24]: Done.

File: docs/SQL/development/target-models/type-and-format.md (hunk 2 above)
Comment (alexanderahn) [reply docs/SQL/development/target-models/type-and-format.md:25]: Done.

File: docs/SQL/development/target-models/write-options.md (hunk 3 above)
Comment (alexanderahn) [reply docs/SQL/development/target-models/write-options.md:24]: Done.

**Pull request merged**